modeltranslation/utils.py

```diff
@@ -13,46 +16,48 @@
     set_enable_fallbacks,
 )
 
+from ._typing import AutoPopulate
+
+_T = TypeVar("_T")
+
 
-def get_language():
+def get_language() -> str | None:
     """
     Return an active language code that is guaranteed to be in
     settings.LANGUAGES (Django does not seem to guarantee this for us).
     """
     lang = _get_language()
-    if lang is None:  # Django >= 1.8
-        return settings.DEFAULT_LANGUAGE
     if lang not in settings.AVAILABLE_LANGUAGES and "-" in lang:
         lang = lang.split("-")[0]
     if lang in settings.AVAILABLE_LANGUAGES:
         return lang
     return settings.DEFAULT_LANGUAGE
 
 
-def get_language_bidi(lang):
+def get_language_bidi(lang: str) -> bool:
     """
     Check if a language is bi-directional.
     """
     lang_info = get_language_info(lang)
     return lang_info["bidi"]
 
 
-def get_translation_fields(field):
+def get_translation_fields(field: str) -> list[str]:
     """
     Returns a list of localized fieldnames for a given field.
     """
     return [build_localized_fieldname(field, lang) for lang in settings.AVAILABLE_LANGUAGES]
 
 
-def build_localized_fieldname(field_name, lang):
+def build_localized_fieldname(field_name: str, lang: str) -> str:
     if lang == "id":
         # The 2-letter Indonesian language code is problematic with the
         # current naming scheme as Django foreign keys also add "id" suffix.
         lang = "ind"
-    return str("%s_%s" % (field_name, lang.replace("-", "_")))
+    return f"{field_name}_{lang.replace('-', '_')}"
```

Comment:
This is also out of scope of the PR.

Upgrading syntax can be handled automatically with pyupgrade, and is not related to types.